va/va_dec_av1.h

```diff
@@ -0,0 +1,546 @@
+            uint8_t         mode_ref_delta_enabled                      : 1;
+            uint8_t         mode_ref_delta_update                       : 1;
+            uint8_t         reserved                                    : 3;
+        } bits;
+        uint8_t             value;
+    } loop_filter_info_fields;
+
+    /** \brief The adjustment needed for the filter level based on
+     *  the chosen reference frame.
+     *  value range [-63..63].
+     */
+    int8_t                  ref_deltas[8];
+
+    /** \brief The adjustment needed for the filter level based on
+     *  the chosen mode.
+     *  value range [-63..63].
+     */
+    int8_t                  mode_deltas[2];
+
+    /** \brief quantization
+     */
+    /** \brief Y AC index
+     *  value range [0..255]
+     */
+    uint16_t                base_qindex;
+    /** \brief Y DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  y_dc_delta_q;
+    /** \brief U DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  u_dc_delta_q;
+    /** \brief U AC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  u_ac_delta_q;
+    /** \brief V DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  v_dc_delta_q;
+    /** \brief V AC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  v_ac_delta_q;
+
+    /** \brief quantization_matrix
+     */
+    union
+    {
+        struct
+        {
+            uint16_t        using_qmatrix                               : 1;
+            /** \brief qm level
+             *  value range [0..15]
+             *  Invalid if using_qmatrix equals 0.
+             */
+            uint16_t        qm_y                                        : 4;
+            uint16_t        qm_u                                        : 4;
+            uint16_t        qm_v                                        : 4;
+            uint16_t        reserved                                    : 3;
+        } bits;
+        uint16_t            value;
+    } qmatrix_fields;
+
+
+    union
+    {
+        struct
+        {
+            /** \brief delta_q parameters
+             */
+            uint32_t        delta_q_present_flag                        : 1;
+            uint32_t        log2_delta_q_res                            : 2;
+
+            /** \brief delta_lf parameters
+             */
+            uint32_t        delta_lf_present_flag                       : 1;
+            uint32_t        log2_delta_lf_res                           : 2;
+
+            /** \brief CONFIG_LOOPFILTER_LEVEL
+             */
+            uint32_t        delta_lf_multi                              : 1;
+
+            /** \brief read_tx_mode
+             *  value range [0..6]
+             */
+            uint32_t        tx_mode                                     : 2;

```

Comment:
how we hold 6 in 2 bits. Is it [0..2]?

Reply:
yes, I will fix it

Reply:
it should be [0..3].

Reply:
we only have following tx mode, right?
0 ONLY_4X4
1 TX_MODE_LARGEST
2 TX_MODE_SELECT

Reply:
you are right,
Value range:
// frame transform mode
typedef enum {
  ONLY_4X4 = 0,     // only 4x4 transform used
  TX_MODE_LARGEST,  // transform size is the largest possible for pu size
  TX_MODE_SELECT,   // transform specified for each block
  TX_MODES,
} TX_MODE;